File: src/Crisp.Ui/ClientApp/src/models/Threat.ts
import { ThreatRecommendation } from "./ThreatRecommendation";

export enum ThreatStatus {
    NotEvaluated,
    NotMitigated,
    PartiallyMitigated,
    Mitigated
}

export enum ThreatRisk {
    NotEvaluated,
    Critical,
    High,
    Medium,
    Low
}
export interface Threat {
    id: string;
    title: string;
    description: string;
    status: ThreatStatus;
    risk: ThreatRisk;
    orderIndex: number;
    recommendations?: ThreatRecommendation[];
    benchmarkIds?: string[];
}

File: src/Crisp.Core/Models/Threat.cs
namespace Crisp.Core.Models;

public enum ThreatStatus
{
    NotEvaluated,
    NotMitigated,
    PartiallyMitigated,
    Mitigated
}

public enum ThreatRisk
{
    NotEvaluated,
    Critical,
    High,
    Medium,
    Low
}

public record Threat(
    string Id,
    string Title,
    string Description,
    ThreatStatus Status,
    ThreatRisk Risk,
    int OrderIndex,
    IEnumerable<ThreatRecommendation>? Recommendations,
    IEnumerable<string>? BenchmarkIds
);
